component/src/main/java/org/ballerinalang/data/cassandra/CassandraDataSource.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.data.cassandra;
+
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.QueryOptions;
+import com.datastax.driver.core.Session;
+import org.ballerinalang.model.types.BType;
+import org.ballerinalang.model.values.BStruct;
+import org.ballerinalang.model.values.BValue;
+
+/**
+ * {@code CassandraDataSource} util class for Cassandra connector initialization.
+ *
+ * @since 0.95.0
+ */
+public class CassandraDataSource implements BValue {
```

Comment:
Why do we need to implement BValue interface here ?

Reply:
Cassandra connector is initialized once in the init method[1] and is kept on the shared map [2] .Later all the other action invocations refer this map to get the data source .In order to add it to the shared map, it needs to be a BValue. 

[1] https://github.com/anupama-pathirage/connector-cassandra/blob/b8a7e76c8ba25b98e3574ce419fc3a4ca3b85872/component/src/main/java/org/ballerinalang/data/cassandra/actions/Init.java#L54
[2] https://github.com/anupama-pathirage/connector-cassandra/blob/b8a7e76c8ba25b98e3574ce419fc3a4ca3b85872/component/src/main/ballerina/ballerina/data/cassandra/natives.bal#L29